### PR TITLE
SSHD-849 working code for pull request

### PR DESCRIPTION
**Pull request by robertodeandrea**
Lyor, this is my pull request according to your instructions for review.

### CODE REVIEW

File: sshd-core/src/main/java/org/apache/sshd/common/forward/DefaultForwardingFilter.java [hunk 1]
```diff
@@ -1082,7 +1093,7 @@ public void messageReceived(IoSession session, Readable message) throws Exceptio
                       session, channel, totalMessages, message.available());
             }
 
-            OpenFuture future = channel.getOpenFuture();
+            OpenFuture future = channel.getPendingMessagesQueue().getCompletedFuture();
```
Comment (lgoldstein) [inline sshd-core/src/main/java/org/apache/sshd/common/forward/DefaultForwardingFilter.java]: Please don't use cascaded calls:
```java
Queue q = channel.getQueue();
Future f = q.getCompletedFuture();
f.addListener(...);
```

**Comment by lgoldstein** [comment]
I don't see any significant changes (perhaps because the PR isn't "clean")

File: sshd-core/src/main/java/org/apache/sshd/client/channel/ClientChannelPendingMessagesQueue.java [hunk 2]
```diff
@@ -39,43 +38,35 @@
...
 
     private final ClientChannel clientChannel;
     private final AtomicBoolean open = new AtomicBoolean(true);
+    private final DefaultOpenFuture completedFuture =
+        new DefaultOpenFuture(ClientChannelPendingMessagesQueue.class.toString(), null);
 
     public ClientChannelPendingMessagesQueue(ClientChannel channel) {
         this.clientChannel = Objects.requireNonNull(channel, "No channel provided");
-        this.completedFuture = new DefaultOpenFuture(getClass().getSimpleName() + "[" + channel + "]", null);
     }
```
[the remark this answers is not in the document]
Comment (robertodeandrea) [reply sshd-core/src/main/java/org/apache/sshd/client/channel/ClientChannelPendingMessagesQueue.java]: Yes, it's a mistake during merge

File: sshd-core/src/main/java/org/apache/sshd/common/forward/DefaultForwardingFilter.java (hunk 1 above)
Comment (fcav1973) [reply sshd-core/src/main/java/org/apache/sshd/common/forward/DefaultForwardingFilter.java]:
```suggestion
            ClientChannelPendingMessagesQueue messagesQueue = channel.getPendingMessagesQueue();
            
            OpenFuture future = messagesQueue.getCompletedFuture();
            
            Consumer<Throwable> errHandler = future.isOpened() ? null : e -> {
                try {
                    exceptionCaught(session, e);
                } catch (Exception err) {
                    log.warn("messageReceived({}) failed ({}) to signal {}[{}] on channel={}: {}",
                        session, err.getClass().getSimpleName(), e.getClass().getSimpleName(),
                        e.getMessage(), channel, err.getMessage());
                }
            };
```

**Pull request closed without merging**